### DISCUSSION TITLE
restrict token connection to main component

### DISCUSSION THREAD

**Discussion by engegu** [six7/figma-tokens#488] [General]
We use Figma Tokens a lot in constructing components for our design system. This means that instances our user pulls from the library has the Figma Tokens "metadata" on them. Is there a way to prevent this?

**Comment by six7** [comment]
Just to understand your use case: You would like the plugin to only detect tokens if they are part of a main component, not if they're part of an instance or on regular layers?

And would this for you be a global setting that you'd set or are there occasions where you do want to make an override on an instance?

**Reply by engegu** [reply]
Yes, the ability to make the instances not inherit assigned tokens from their main components would make it a lot cleaner for us.  I imagine there would be performance gains too? Alternatively a «remove tokens when shared to other files as a library assets» setting.

**Reply by engegu** [reply]
They would still get their values from the main component  ofc (which gets them from tokens), just not any token connection.

**Reply by engegu** [reply]
You could still apply tokens to these «clean» instances,  no reason to restrict this.

**Reply by engegu** [reply]
The more i think about it, the more i am convinced a simple "prevent token inheritance" global setting would be fantastic :D

**Reply by six7** [reply]
Yeah we'd need this to be global, otherwise we'd need to store a decision like that in each token. What if it allowed you to disable token inheritance, then tokens inside of component instances would be ignored, could also speed up the plugin a bit as it wouldn't have to traverse the tree further down if it encounters an instance, it would just skip to the next layer(s)
